### PR TITLE
Support IG backends removal in regional_ig_linker.go.

### PR DESCRIPTION
**Pull request by mmamczur**
Tried to make it so that whatever exists in the backends list is preserved instead of overwriting with 'wantIGs'.

### CODE REVIEW

File: pkg/backends/ig_linker_test.go [hunk 1]
```diff
@@ -251,15 +260,22 @@ func TestGetIGLinksToAdd(t *testing.T) {
...
+				t.Fatalf("getInstanceGroupsToAddAndRemove(_,_): err:%v ", err)
 			}
 			sort.Slice(toAdd, func(i, j int) bool {
 				return toAdd[i] <= toAdd[j]
 			})
-			if !reflect.DeepEqual(toAdd, tc.want) {
-				t.Fatalf("getInstanceGroupsToAdd(_,_) error. Got:%v, Want:%v", toAdd, tc.want)
+			sort.Slice(tc.wantRemove, func(i, j int) bool {
+				return tc.wantRemove[i] <= tc.wantRemove[j]
+			})
```
Comment (panslava) [inline pkg/backends/ig_linker_test.go:272]: we can just do sort.Strings

Comment (panslava) [reply pkg/backends/ig_linker_test.go:272]: and on the previous lines too?

Comment (mmamczur) [reply pkg/backends/ig_linker_test.go]: done

**Comment by panslava** [comment]
code looks good to me, but still want @swetharepakula to take a look, I am not sure, if we should remove instance groups, and why it was not done before (in ig_linker.go as well)

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**